apps/common-app/src/examples/RuntimeTests/ReJest/TestRunner/TestRunner.ts

```diff
@@ -68,142 +70,54 @@ export class TestRunner {
-      componentsRefs: {},
-      callsRegistry: {},
-      errors: [],
-      skip: decorator === TestDecorator.SKIP || this._currentTestSuite.decorator === DescribeDecorator.SKIP,
-      decorator,
-    });
-  }
-
-  public testEach<T>(examples: Array<T>, decorator: TestDecorator | null) {
-    return (name: string, testCase: (example: T, index: number) => void | Promise<void>) => {
-      examples.forEach((example, index) => {
-        const currentTestCase = async () => {
-          await testCase(example, index);
-        };
-        this.test(formatTestName(name, example, index), currentTestCase, decorator);
-      });
-    };
-  }
-
-  public useTestRef(name: string): MutableRefObject<Component | null> {
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    const ref = useRef(null);
-    assertTestCase(this._currentTestCase);
-    this._currentTestCase.componentsRefs[name] = ref;
-    return ref;
-  }
-
-  public getTestComponent(name: string): TestComponent {
-    assertTestCase(this._currentTestCase);
-    const componentRef = this._currentTestCase.componentsRefs[name];
-    return new TestComponent(componentRef);
-  }
-
   public async runTests() {
     console.log('\n');
+    await this._testSuiteInitialManager.buildSuites();
+    this._testSuiteInitialManager.setSkipFlags();
 
-    for (const testSuite of this._testSuites) {
-      this._currentTestSuite = testSuite;
-      await testSuite.buildSuite();
-      this._currentTestSuite = null;
-    }
-
-    for (const testSuite of this._testSuites) {
-      let skipTestSuite = testSuite.skip;
+    const testSuiteExecutionManager = new TestSuiteExecutionManager(this._testSuiteInitialManager);
+    this._testSuiteExecutionManager = testSuiteExecutionManager;
 
-      if (this._includesOnly) {
-        skipTestSuite = skipTestSuite || !(testSuite.decorator === DescribeDecorator.ONLY);
-
-        for (const testCase of testSuite.testCases) {
-          if (testCase.decorator === TestDecorator.ONLY) {
-            skipTestSuite = false;
-          } else {
-            testCase.skip = testCase.skip || !(testSuite.decorator === DescribeDecorator.ONLY);
-          }
-        }
-      }
-      testSuite.skip = skipTestSuite;
+    for (const testSuite of testSuiteExecutionManager.getTestSuites()) {
+      await this.runTestSuite(testSuite, testSuiteExecutionManager);
     }
 
-    for (const testSuite of this._testSuites) {
-      await this.runTestSuite(testSuite);
-    }
-
-    this._testSuites = [];
     this._testSummary.printSummary();
   }
 
-  private async runTestSuite(testSuite: TestSuite) {
+  private async runTestSuite(testSuite: TestSuite, testSuiteExecutionManager: TestSuiteExecutionManager) {
     this._testSummary.countSkippedTestSuiteTests(testSuite);
-
     if (testSuite.skip) {
       return;
     }
 
-    this._currentTestSuite = testSuite;
     this._testSummary.logRunningTestSuite(testSuite);
+    testSuiteExecutionManager.setCurrentTestSuite(testSuite);
```

Comment:
This is confusing. Why do we need to pass a test suite to the test suite manager? This seems like a mixed responsibility.